### PR TITLE
Add a script to make testing upgrades easier.

### PR DESCRIPTION
**Pull request by SpicyLemon**
## Description

Add a script to make testing upgrades easier.
Update the documentation to direct people to the script instead of the several steps the script takes care of.

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote unit and integration [tests](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#testing)
- [x] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Comment by iramiller** [comment]
When I test this I have issues with the command line argument method of creating a proposal...

```
++ build/provenanced tx gov submit-proposal software-upgrade brown --title 'Upgrade for brown' --description 'Upgrading provenance to brown' --upgrade-info=brown --from tp1gkcqyp67ad2y668us4lpqp0053ctjjm6jrqypm --upgrade-height 64 --deposit 10000000nhash --chain-id=testing --keyring-backend test --gas-prices=1905nhash --gas=auto --gas-adjustment=1.5 --yes
Error: unknown flag: --title
Usage:
  provenanced tx gov submit-proposal [path/to/proposal.json] [flags]
```

**Comment by SpicyLemon** [comment]
> When I test this I have issues with the command line argument method of creating a proposal...
> 
> ```
> ++ build/provenanced tx gov submit-proposal software-upgrade brown --title 'Upgrade for brown' --description 'Upgrading provenance to brown' --upgrade-info=brown --from tp1gkcqyp67ad2y668us4lpqp0053ctjjm6jrqypm --upgrade-height 64 --deposit 10000000nhash --chain-id=testing --keyring-backend test --gas-prices=1905nhash --gas=auto --gas-adjustment=1.5 --yes
> Error: unknown flag: --title
> Usage:
>   provenanced tx gov submit-proposal [path/to/proposal.json] [flags]
> ```

Hah! Crap. I tested this script running v1.12. But with Cosmos v0.46, they overhauled gov proposals and this command changed.

Ugh. I'm not sure of how to best detect and handle that. I guess I could run the `--help` on it and grep it to test the version. Might be overkill though. Basically, from 1.12.x to 1.13.0, we'll need this version, then the new version after that. But if we have a few rcs in testnet before 1.13.0, we'll want both versions....

Looking into it a bit more.

**Comment by iramiller** [comment]
I think maybe building up some json and using that would work with all versions ... I always put gov proposals in that way and haven't ever used the command line flags...

File: scripts/upgrade-test.sh [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+set -ex
+export PIO_TESTNET=true
+export PIO_HOME="${PIO_HOME:-build/run/provenanced}"
+export PIO_OUTPUT=json
+
+valAddr="$( "$prov" keys list --output json | jq -r '.[] | select( .name == "validator" ) | .address' )"
+curHeight="$( "$prov" status | jq -r '.SyncInfo.latest_block_height' )"
+targetHeight="$(( curHeight + 40 ))"
+
+propRes="$( "$prov" tx gov submit-proposal software-upgrade "$color" \
```
Comment (Taztingo) [inline scripts/upgrade-test.sh:26]: Might be helpful to have chain-id as configurable. I've had to update multiple local chains at the same time. Also, might be helpful to have deposit and gas-prices configurable.

File: scripts/upgrade-test.sh [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    --chain-id=testing \
+    --keyring-backend test \
+    --gas-prices='1905nhash' \
+    --gas=auto \
+    --gas-adjustment=1.5 \
+    --yes )"
+
+propId="$( tail -n 1 <<< "$propRes" | jq -r '.logs[0].events[] | select( .type == "submit_proposal" ) | .attributes[] | select( .key == "proposal_id" ) | .value' )"
+
+"$prov" tx gov vote "$propId" yes \
```
Comment (Taztingo) [inline scripts/upgrade-test.sh:42]: Same as above.

File: scripts/upgrade-test.sh (hunk 1 above)
Comment (SpicyLemon) [reply scripts/upgrade-test.sh:26]: I've revamped the script so it's all configurable through env vars.

Comment (SpicyLemon) [reply scripts/upgrade-test.sh]: Sorry,... forgot to push. Updated for reals now.

File: docs/testing_software_upgrade.md [hunk 3]
```diff
@@ -8,65 +8,22 @@ This document outlines how to do a local software upgrade for testing purposes o
 An upgrade in Provenance is done by a node submitting a proposal with a deposit in hash.  This proposal contains the blockchain height as well as the upgrade name.  Once a sufficient deposit has been made the proposal enters a voting period where any user with hash may vote.  After the voting period, defined in the genesis.json, the proposal will either pass or fail.  If the proposal passes, the chain will stop at the height of the proposal.  Then the chain must be restarted with the upgrade handler with the upgrade name.  This upgrade handler will execute and the chain will continue running with the software upgrade.
 
 ## Steps
-1. [ ] Check out previous version of Provenance
+1. [ ] Check out previous version of Provenance.
     - For example: `git checkout v1.7.5`
-   
-2. [ ] Build and configure Provenance
+
+2. [ ] Build and configure Provenance:
```
[the remark this answers is not in the document]
Comment (SpicyLemon) [reply docs/testing_software_upgrade.md:14]: Maybe need `go mod vendor`? Or maybe you're on another branch? This PR doesn't change any `go` stuff, and doesn't affect the build at all.

**Review by arnabmitra — APPROVED** [review]

**Review by Taztingo — APPROVED** [review]
LGTM

**Pull request merged**